apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -53,5 +74,46 @@ func validateRAID(r *RAIDConfig) error {
 		return fmt.Errorf("hardwareRAIDVolumes and softwareRAIDVolumes can not be set at the same time")
 	}
 
+	if bmcAccess == nil {
+		return nil
+	}
+
+	if bmcAccess.RAIDInterface() == "no-raid" {
```

Comment:
Minor: wouldn't be better to do this check as first?

Reply:
yes, updated. PTAL.